### PR TITLE
Release/fix 09citizen avatar widget test

### PR DESCRIPTION
**Pull request by fryyd**
Intergration tests on the avatar widget.

### CODE REVIEW

**Comment by User12837985123** [comment]
@aau-giraf/sw602f19 you have been assigned this. Here's your checklist. I marked @Jacob6565 as the reviewer, but you might change it as you like.
Please use the following checklist. Feel free to check it off as you go along.
**Code Design**
- [ ] Is the code in the right place? (Both in terms of folder structure and class structure)
- [ ] Could this code, to the best of your knowledge, have reused existing code?
- [ ] Is the code over-engineered? Examples of over-engineering: Implemented behavior for future problems
- [ ] Does it introduce functionality that is not necessary for solving the problem?

**Code Readability and Maintainability**
- [ ] Are names meaningful and self-documenting?
- [ ] Is it understandable, by reading the code, what it does?
- [ ] Is the code simple and readable? i.e. it contains no “hack”, or obscure solution.
- [ ] Is it understandable, by reading the tests, what they do?
- [ ] Do the tests cover sensible cases? Both happy paths and exception paths?
- [ ] Do the tests cover the full functionality? i.e., will the tests fail if some of the requested functionality is missing?
- [ ] Are error messages in the code understandable?
- [ ] Are confusing sections of code covered by comments?

**Functionality**
For this part, open the app, and test:
- [ ] Are the features that the code claims to implement, actually implemented?
- [ ] Does it look like the code has bugs?
- [ ] Are all new screens reachable through in-app navigation?

**Points for Discussion and Reflection**
- [ ] Are UI messages, that the user might meet, actually helpful?
- [ ] Are there errors that are bad practice? Like, hard-coding a test database connection?
- [ ] Is there well written code in this? Remember to tell the developer!

**Comment by User12837985123** [comment]
@aau-giraf/sw608f19 you have been assigned this. Here's your checklist. I marked @TrineHolmager as the reviewer, but you might change it as you like.
Please use the following checklist. Feel free to check it off as you go along.
**Code Design**
- [ ] Is the code in the right place? (Both in terms of folder structure and class structure)
- [ ] Could this code, to the best of your knowledge, have reused existing code?
- [ ] Is the code over-engineered? Examples of over-engineering: Implemented behavior for future problems
- [ ] Does it introduce functionality that is not necessary for solving the problem?

**Code Readability and Maintainability**
- [ ] Are names meaningful and self-documenting?
- [ ] Is it understandable, by reading the code, what it does?
- [ ] Is the code simple and readable? i.e. it contains no “hack”, or obscure solution.
- [ ] Is it understandable, by reading the tests, what they do?
- [ ] Do the tests cover sensible cases? Both happy paths and exception paths?
- [ ] Do the tests cover the full functionality? i.e., will the tests fail if some of the requested functionality is missing?
- [ ] Are error messages in the code understandable?
- [ ] Are confusing sections of code covered by comments?

**Functionality**
For this part, open the app, and test:
- [ ] Are the features that the code claims to implement, actually implemented?
- [ ] Does it look like the code has bugs?
- [ ] Are all new screens reachable through in-app navigation?

**Points for Discussion and Reflection**
- [ ] Are UI messages, that the user might meet, actually helpful?
- [ ] Are there errors that are bad practice? Like, hard-coding a test database connection?
- [ ] Is there well written code in this? Remember to tell the developer!

File: lib/widgets/citizen_avatar_widget.dart [hunk 1]
```diff
@@ -27,6 +27,7 @@ class CitizenAvatar extends StatelessWidget {
                             aspectRatio: 1,
                             child: Container(
                               child: const CircleAvatar(
+                                key: Key('WidgetAvatar'),
```
Comment (User12837985123) [inline lib/widgets/citizen_avatar_widget.dart:45]: This is not a comment to this line, but I noticed that there are some documentation missing in this file, specfically documentation of the class, contructor and it's public members.

File: test/widgets/citizen_avatar_widget_test.dart [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+}
+
+void main() {
+  int i = 0;
+  final MockScreen mockScreen = MockScreen(callback: () => ++i);
+
+  testWidgets('Test if citizen text appears', (WidgetTester tester) async {
+    await tester.pumpWidget(MaterialApp(home: mockScreen));
+    await tester.pumpAndSettle();
+    expect(find.byKey(const Key('WidgetText')), findsNWidgets(1));
```
Comment (User12837985123) [inline test/widgets/citizen_avatar_widget_test.dart]: findsNWidgets(1) is equivalent to findsOneWidget, right? Just have to make sure :D

**Review by TrineHolmager — DISMISSED** [review]

File: test/widgets/citizen_avatar_widget_test.dart [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  });
+
+  testWidgets('Test if citizen avatar appears', (WidgetTester tester) async {
+    await tester.pumpWidget(MaterialApp(home: mockScreen));
+    await tester.pumpAndSettle();
+    expect(find.byKey(const Key('WidgetAvatar')), findsNWidgets(1));
```
Comment (User12837985123) [inline test/widgets/citizen_avatar_widget_test.dart]: Wouldn't it be better to make the key to a variable and then use it instead.

**Review by User12837985123 — DISMISSED** [review]

**Review by TrineHolmager — DISMISSED** [review]

**Review by User12837985123 — APPROVED** [review]

**Review by Eduardsen — APPROVED** [review]

**Pull request merged**